pkg/apis/navigator/validation/elasticsearch_test.go

```diff
@@ -472,3 +468,62 @@ func TestValidateElasticsearchClusterSpec(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateElasticsearchClusterUpdate(t *testing.T) {
+	type testT struct {
+		old           *navigator.ElasticsearchCluster
+		new           *navigator.ElasticsearchCluster
+		errorExpected bool
+	}
+
+	baseCluster := validESCluster.DeepCopy()
+	baseCluster.Spec.NodePools = []navigator.ElasticsearchClusterNodePool{
+		newValidNodePool("test", 3, navigator.ElasticsearchRoleMaster),
+	}
+
+	setPersistence := func(
+		e *navigator.ElasticsearchCluster,
+		p navigator.PersistenceConfig,
+	) *navigator.ElasticsearchCluster {
+		e = e.DeepCopy()
+		e.Spec.NodePools[0].Persistence = p
+		return e
+	}
+
+	tests := map[string]testT{
+		"unchanged cluster": {
+			old: baseCluster,
+			new: baseCluster,
+		},
+		"enable persistence config": {
+			old: setPersistence(baseCluster, navigator.PersistenceConfig{Enabled: false}),
+			new: baseCluster,
+		},
+	}
+
+	for title, persistence := range persistenceErrorCases {
+		tests[title] = testT{
+			old:           baseCluster,
+			new:           setPersistence(baseCluster, persistence),
+			errorExpected: true,
+		}
+	}
+
+	for title, tc := range tests {
+		t.Run(
+			title,
+			func(t *testing.T) {
+				errs := validation.ValidateElasticsearchClusterUpdate(tc.old, tc.new)
+				if tc.errorExpected && len(errs) == 0 {
+					t.Errorf("expected error but got none")
+				}
+				if !tc.errorExpected && len(errs) != 0 {
+					t.Errorf("unexpected errors: %s", errs)
+				}
+				for _, e := range errs {
+					t.Logf("error string is: %s", e)
+				}
+			},
+		)
+	}
+}
```

Comment:
I ran the tests locally and couldn't see the equivalent for ElasticSearch of:

```
    --- PASS: TestValidateCassandraClusterUpdate/changed_rack (0.00s)
    	cassandra_test.go:172: error string is: spec: Forbidden: updates to nodepool for fields other than 'replicas' and 'persistence' are forbidden.
```

Do we need an extra test for changing something other than persistence on  `ElasticSearchNodePool`
Ignore me if you think this is nit-picking.

Reply:
Sure, I'll add extra test coverage